lib/Doctrine/DBAL/Driver/SQLAnywhere/SQLAnywhereConnection.php

```diff
@@ -137,7 +137,13 @@ public function lastInsertId($name = null)
             return sasql_insert_id($this->connection);
         }
 
-        return $this->query('SELECT ' . $name . '.CURRVAL')->fetchColumn();
+        $value = $this->query('SELECT ' . $name . '.CURRVAL')->fetchColumn();
+
+        if ($value === false) {
+            return '';
+        }
```

Comment:
Yeah, this should be suppressed. I'm not sure if the “BC break” is the right term here because the new value corresponds to the documented return type while the old one didn't.